x-pack/plugins/cases/server/saved_object_types/migrations/comments.ts

```diff
@@ -103,33 +105,43 @@ export const createCommentsMigrations = (
   return mergeMigrationFunctionMaps(commentsMigrations, embeddableMigrations);
 };
 
-const migrateByValueLensVisualizations =
-  (migrate: MigrateFunction, version: string): SavedObjectMigrationFn<{ comment?: string }> =>
-  (doc: SavedObjectUnsanitizedDoc<{ comment?: string }>) => {
+export const migrateByValueLensVisualizations =
+  (
+    migrate: MigrateFunction,
+    version: string
+  ): SavedObjectMigrationFn<{ comment?: string }, { comment?: string }> =>
+  (doc: SavedObjectUnsanitizedDoc<{ comment?: string }>, context: SavedObjectMigrationContext) => {
     if (doc.attributes.comment == null) {
       return doc;
     }
 
-    const parsedComment = parseCommentString(doc.attributes.comment);
-    const migratedComment = parsedComment.children.map((comment) => {
-      if (isLensMarkdownNode(comment)) {
-        // casting here because ts complains that comment isn't serializable because LensMarkdownNode
-        // extends Node which has fields that conflict with SerializableRecord even though it is serializable
-        return migrate(comment as SerializableRecord) as LensMarkdownNode;
-      }
+    try {
+      const parsedComment = parseCommentString(doc.attributes.comment);
+      const migratedComment = parsedComment.children.map((comment) => {
+        if (isLensMarkdownNode(comment)) {
+          // casting here because ts complains that comment isn't serializable because LensMarkdownNode
+          // extends Node which has fields that conflict with SerializableRecord even though it is serializable
+          return migrate(comment as SerializableRecord) as LensMarkdownNode;
+        }
 
-      return comment;
-    });
+        return comment;
+      });
 
-    const migratedMarkdown = { ...parsedComment, children: migratedComment };
+      const migratedMarkdown = { ...parsedComment, children: migratedComment };
 
-    return {
-      ...doc,
-      attributes: {
-        ...doc.attributes,
-        comment: stringifyCommentWithoutTrailingNewline(doc.attributes.comment, migratedMarkdown),
-      },
-    };
+      return {
+        ...doc,
+        attributes: {
+          ...doc.attributes,
+          comment: stringifyCommentWithoutTrailingNewline(doc.attributes.comment, migratedMarkdown),
+        },
+      };
+    } catch (error) {
+      // TODO: we don't have access to the context (it's always undefined because of the way the migrations are merged)
+      // so we can't log anything
+      // logError({ id: doc.id, context, error, docType: 'comment', docKey: 'comment' });
```

Comment:
`context` is always `undefined` so we're probably going to need to remove the log for now, unless I'm missing something